java/integTest/src/test/java/glide/cluster/ClusterTransactionTests.java

```diff
@@ -83,6 +85,36 @@ public void info_simple_route_test() {
         assertTrue(((String) result[1]).contains("# Stats"));
     }
 
+    @SneakyThrows
+    @Test
+    public void test_xread_transactions() {
+        String key9 = "{key}" + UUID.randomUUID();
+
+        ClusterTransaction transaction =
+                new ClusterTransaction()
+                        .xadd(key9, Map.of("field1", "value1"), StreamAddOptions.builder().id("0-1").build())
+                        .xadd(key9, Map.of("field2", "value2"), StreamAddOptions.builder().id("0-2").build())
+                        .xadd(key9, Map.of("field3", "value3"), StreamAddOptions.builder().id("0-3").build())
+                        .xread(Map.of(key9, "0-1"));
+
+        Object[] results = clusterClient.exec(transaction, RANDOM).get();
+
+        // verify xadd results
+        assertEquals("0-1", results[0]);
+        assertEquals("0-2", results[1]);
+        assertEquals("0-3", results[2]);
+
+        @SuppressWarnings("unchecked")
+        Map<String, Map<String, Map<String, String>>> xreadResult =
+                (Map<String, Map<String, Map<String, String>>>) results[3];
+        assertEquals(Set.of("0-2", "0-3"), xreadResult.get(key9).keySet());
```

Comment:
Why you can't add to our common transaction test?

Reply:
Now that I've added the ID option, it works in the transaction test.  So I can remove this.